### ISSUE TITLE
Turn on location dialog should turn on GPS directly by clicking YES

### ISSUE THREAD

**Issue by Ayan-10** [commons-app/apps-android-commons#4320]
**Summary:** 

If GPS is not turned on and user click YES in the Turn on location dialog. It should turn on GPS directly instead of going to Settings like Google Maps.

**Steps to reproduce:** 
* Turn off GPS in your device.
* Open the app and go to Nearby
* Click Recenter FAB
* Click YES in Turn on location dialog

**What did you expect the app to do, and what did you see instead?**
The app should directly turn on GPS after clicking YES instead of going to settings.

**Device and Android version:** 

Device: Pixel 3a
API: 27
 
**Commons app version:** 
3.0.0-debug-helpLink~afd977f6c

**Screen-shots:** 
<table>
  <tr>
    <td> Dialog</td>
     <td> After clicking YES</td>
  </tr>
  <tr>
    <td><img src="https://user-images.githubusercontent.com/71203077/113499463-ca030b80-9533-11eb-9a6a-27525a7a10ee.png"width="250"></td>
    <td><img src="https://user-images.githubusercontent.com/71203077/113499487-ebfc8e00-9533-11eb-851b-6be728c99a1b.png"width="250"></td>
  </tr>
 </table>

**Would you like to work on the issue?**
**Yes, I would like to work on this. @nicolas-raoul Can you kindly assign me?**

**Comment by nicolas-raoul** [comment]
Indeed Google Maps turns the "Location" setting on directly, rather than directing the user to the settings screen.
Thanks a lot @Ayan-10 for finding this, but again this would be an enhancement rather than a bug (the feature currently works but could be made more convenient).

**Comment by Ayan-10** [comment]
Yeah right, Ok no problem. @nicolas-raoul Actually recently I made an App where I implemented a similar idea. So I was thinking If you can assign me to this issue or not. But if it's a protocol then it's ok.

**Comment by Ayan-10** [comment]
@nicolas-raoul can I consider it for GSOC?

**Comment by nicolas-raoul** [comment]
@Ayan-10 Thanks for asking! I also think this one would be too simple, actually.

**Comment by Ayan-10** [comment]
@nicolas-raoul ok I can understand. Actually, this is my first time applying for GSOC.

**Comment by obsidian-zero** [comment]
I'm sorry, but from what I remember.
After Android 5.0 normal circumstances can not directly open the GPS through the software.
Did I remember wrong?
